### PR TITLE
update approach and fixed issues #64, #65 and #71

### LINKED ISSUE #64: After updating to 1.4.0+ translates don't refresh immediately after setLocale().
I have 4 languages in app.
After update on 1.4.0 and 1.4.1 the tr() methods not update the texts after setLocale() is called. Texts updates only after change screen to another. Even if i use StreamBuilder and refresh page by itself texts not changes.

But, when i change (for example) from EN to RU (nothing is updates) but if after that i change language from RU to FR (texts updates to RU but locale setted to FR) and next if change back to EN (text updates to FR). I dont know how explain it clearer.
Maybe a "key" property helps?

P.S. On version 1.3.5 all works fine and languages updates immediately after setLocale().

### LINKED ISSUE #71: convert ensureInitialized() to internal func
@danilofuchs @spiritinlife I think would be best if we hiding `ensureInitialized()` from `main() ` and make `main()`  without `asycn` like a default 
[Good or Bad: Declaring main method async in Dart / Flutter](https://stackoverflow.com/questions/56129121/good-or-bad-declaring-main-method-async-in-dart-flutter)
```dart
void main() {
  runApp(EasyLocalization(
    child: MyApp(),
    supportedLocales: [Locale('en', 'US'), Locale('ar', 'DZ')],
    path: 'resources/langs',
  ));
}
```

### PR DESCRIPTION
**Pull request by aissat**
- make 'fallbackLocale' optional arguments ,if it is null, then you can take it from phone default localization #65 
- fixed #71  removed `async` from `main()` and hiding  `ensureInitialized` from `main()`
- fixed #64 don't refresh immediately
- added code coverage info badges 
- updated example

### CODE REVIEW

File: example/ios/Runner.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -17,6 +17,7 @@
 		97C146FC1CF9000F007C117D /* Main.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = 97C146FA1CF9000F007C117D /* Main.storyboard */; };
 		97C146FE1CF9000F007C117D /* Assets.xcassets in Resources */ = {isa = PBXBuildFile; fileRef = 97C146FD1CF9000F007C117D /* Assets.xcassets */; };
```
Comment (spiritinlife) [inline example/ios/Runner.xcodeproj/project.pbxproj]: Should this be added to gitingore?

File: example/lib/lang_view.dart [hunk 2]
```diff
@@ -0,0 +1,104 @@
+import 'package:flutter/material.dart';
+import 'package:easy_localization/easy_localization.dart';
+
+class LanguageView extends StatefulWidget {
```
Comment (spiritinlife) [inline example/lib/lang_view.dart]: This could be a statelesswidget

File: example/lib/lang_view.dart [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+          // isThreeLine: true,
+          title: Text(
+            title,
+          ),
+          subtitle: Text(
+            subtitle,
+          ),
+          onTap: () {
+            setState(() {
+              EasyLocalization.of(context).locale =locale;
```
Comment (spiritinlife) [inline example/lib/lang_view.dart]: No need to call setState

File: example/lib/main.dart [hunk 4]
```diff
@@ -84,29 +85,15 @@ class _MyHomePageState extends State<MyHomePage> {
...
+              );
             },
           ),
-          FlatButton(
-            child: Text("عربي"),
-            color: Localizations.localeOf(context).languageCode == "ar"
-                ? Colors.lightBlueAccent
-                : Colors.blue,
-            onPressed: () {
-              this.setState(() {
```
Comment (spiritinlife) [inline example/lib/main.dart]: No need to call setState

File: lib/src/easy_localization_app.dart [hunk 5]
```diff
@@ -43,11 +43,15 @@ class EasyLocalization extends StatefulWidget {
...
   static Locale _savedLocale;
+  // Get default OS Locale
+  static final _osCurrentLocale = Intl.getCurrentLocale().split("_");
 
   // @TOGO maybe add assertion to ensure that ensureInitialized has been called and that
   // _savedLocale is set.
   _EasyLocalizationLocale(Locale fallbackLocale)
-      : this._locale = _savedLocale ?? fallbackLocale;
+      // if _savedLocale and fallbackLocale null init by default OS Locale
+      : this._locale = (_savedLocale ?? fallbackLocale) ??
```
Comment (spiritinlife) [inline lib/src/easy_localization_app.dart]: If the device local is not in supportedLocals and language files do not exist for this locale, what will happen? 
I think localization will fail no?

File: example/ios/Runner.xcodeproj/project.pbxproj (hunk 1 above)
Comment (danilofuchs) [reply example/ios/Runner.xcodeproj/project.pbxproj]: Actually not! This looks auto-generated, but there are some fields we might want to manually edit.
https://github.com/flutter/flutter/issues/12639#issuecomment-338037309

**Comment by aissat** [comment]
> We came into the conclusion that we should use Flutter's built-in locale resolution mechanism, as it was before the changes introduced in #66. So, `fallbackLocale` should be removed, along with `localeResolutionCallback`.

@danilofuchs  @spiritinlife  I don't really go in for removed  `fallbackLocale` coz :
 - I think the default needs to select a Locale from the list "like English as default" to start the first time.
- sometimes when building an app needs this app to take the same  default os locale in this make `fallbackLocale` optional arguments

File: lib/src/easy_localization_app.dart (hunk 5 above)
Comment (aissat) [reply lib/src/easy_localization_app.dart]: we can resolve by taking from `supportedLocales`

```dart
// if _savedLocale and fallbackLocale null and default OS Locale in supportedLocales 
  // init by default OS Locale else init by supportedLocales[0]
      : this._locale = (_savedLocale ?? fallbackLocale) ??
      supportedLocales.firstWhere((local) => local ==_osLocal, orElse: () => supportedLocales.first); 
```

File: lib/src/easy_localization_app.dart [hunk 6]
```diff
@@ -95,7 +96,10 @@ class _EasyLocalizationState extends State<EasyLocalization> {
 
   @override
   void initState() {
-    _locale = _EasyLocalizationLocale(widget.fallbackLocale);
+    _locale =
+        _EasyLocalizationLocale(widget.fallbackLocale, this.supportedLocales);
+    _locale.initSavedAppLocale();
```
Comment (spiritinlife) [inline lib/src/easy_localization_app.dart]: @aissat by doing this here you will cause a flicker if the user has selected a locale the previous the app was open that is different from the device locale. That is why i was doing this ensureInitialized call in main

Comment (spiritinlife) [reply lib/src/easy_localization_app.dart]: By flicker i mean that text will be initialy in the device locale and then change to the previous saved locale

Comment (aissat) [reply lib/src/easy_localization_app.dart]: I don't noticed this, everything works well but really great **note** it true

> text will be initialy in the device locale and then change to the previous saved locale

**Pull request merged**